server/src/main/java/org/elasticsearch/common/bytes/ReleasableBytesReference.java

```diff
@@ -129,6 +130,7 @@ public long ramBytesUsed() {
     @Override
     public StreamInput streamInput() throws IOException {
         assert hasReferences();
+        // NOCOMMIT why doesn't this inc the ref?
```

Comment:
There's no particular reason really and I agree that this might not be optimal. We could make it increment the ref (and decrement on close) and I bet we could use that to clean up quite a bit of code but all the existing code is written under the assumption that all the lifecycle lives with the bytes reference and that's a bigger task to change without introducing leaks I think.

Reply:
Yeah - I don't want to change this as part of this PR, but it sure confused the hell out of me.